### PR TITLE
Add Xata guide to BE Recipe section

### PR DESCRIPTION
**Pull request by atilafassina**
Hey folks, here I am again 🥳
Trying to fill up that stub for Xata as I promised.

## What kind of changes does this PR include?
-  Updated content

## Description

Adds a quick start guide with Xata.

## Changed route

- [Backend Service Guide: Xata](https://deploy-preview-3207--astro-docs-2.netlify.app/en/guides/backend/xata/)

### CODE REVIEW

File: src/content/docs/en/guides/backend/xata.mdx
Comment (ElianCodes) [inline src/content/docs/en/guides/backend/xata.mdx:60]:
```suggestion
# the same name as your git branch
```

**Review by ElianCodes — COMMENTED** [review]
Just a small suggestion; but overall LGTM!

**Review by ElianCodes — APPROVED** [review]
Looks awesome! It's a merge for me! @sarah11918 or @Yan-Thomas, any thoughts?

File: src/content/docs/en/guides/backend/xata.mdx
Comment (Yan-Thomas) [inline src/content/docs/en/guides/backend/xata.mdx:92]: I believe here the preposition should be "in" instead.
```suggestion
Additionally, the prompt may have assisted you in creating an instance of the SDK (Software Development Kit) and, if you chose TypeScript, you also have the types generated from your schema. Notice `@xata.io/client` was added to your `package.json` as well.
```

**Review by Yan-Thomas — COMMENTED** [review]
Nice work @atilafassina, thanks 🙌. I have one small nit, overall LGTM! Let's see Sarah's thoughts before merging 😄

File: src/content/docs/en/guides/backend/xata.mdx
Comment (atilafassina) [reply src/content/docs/en/guides/backend/xata.mdx:92]: Great catch!! Thanks!

**Review by sarah11918 — CHANGES_REQUESTED** [review]
Thanks for coming back and contributing, @atilafassina ! Looking forward to having this in docs!

Just leaving this note here because I *will* be editing this, but it might not be until next week. And this is just a note to the rest of the team that this shouldn't be merged before I do!  :rocket:

**Comment by sarah11918** [comment]
Well, looks like my projections were a *little* optimistic!  :sweat_smile: 

Just popping in to say that Kevin and/or I will be getting around to this one at some point this week! Thanks for hanging in there!  :raised_hands:

**Comment by atilafassina** [comment]
> Well, looks like my projections were a *little* optimistic!  :sweat_smile: 
> 
> 
> 
> Just popping in to say that Kevin and/or I will be getting around to this one at some point this week! Thanks for hanging in there!  :raised_hands: 

thank you, Sarah!
Let me know if there's anything I can do to help! 😊

File: src/content/docs/en/guides/backend/xata.mdx
Comment (sarah11918) [inline src/content/docs/en/guides/backend/xata.mdx:12]:
```suggestion
[Xata](https://xata.io) is a **Serverless Data Platform** that combines the features of a relational database, a search engine, and an analytics engine by exposing a single consistent REST API.
```

File: src/content/docs/en/guides/backend/xata.mdx
Comment (sarah11918) [inline src/content/docs/en/guides/backend/xata.mdx:18]:
```suggestion
- A [Xata](https://app.xata.io/signin) account with a created database. (You can use the sample database from the Web UI.)
```

File: src/content/docs/en/guides/backend/xata.mdx
Comment (sarah11918) [inline src/content/docs/en/guides/backend/xata.mdx:21]: So, I just want to be clear here, because we normally ONLY include the stuff required to make 3rd parties work with Astro: Is this environment config an *Astro-specific* configuration step, or is this something that Xata docs describe how to do to have Xata working with any project?

My gut feeling is that this is not something for this guide, but something that could be added as a pre-requisite line (i.e. Xata CLI configured globally, maybe with a link to Xata docs for this?)

These guides can get pretty long, and we try to include as little as possible that's (a) documented elsewhere and (b) subject to change in someone else's project that we wouldn't necessarily know about.

File: src/content/docs/en/guides/backend/xata.mdx
Comment (sarah11918) [inline src/content/docs/en/guides/backend/xata.mdx:64]: This seems to me to be the important part of environment config, for an Astro project specifically? Maybe this is the only thing in this section (assuming it's easy enough to include setting up your environment as a prerequisite?)

Note: I might be wrong! You might need this entire section in order to integrate with Astro. If so, do let me know! I just know it's pretty common for people to PR guides that include setting up their *own* parts of the 3rd party, which is usually explained and doc'ed elsewhere. So, I need to check!  :smile:

File: src/content/docs/en/guides/backend/xata.mdx
Comment (sarah11918) [inline src/content/docs/en/guides/backend/xata.mdx:108]:
```suggestion
To query your posts, import and use `getXataClient()` in a `.astro` file. The example below queries the first 50 posts from Xata's Sample Blog Database.
```
Just editing for tone/voice here, as per our Writing/style guide!

File: src/content/docs/en/guides/backend/xata.mdx
Comment (sarah11918) [inline src/content/docs/en/guides/backend/xata.mdx:129]:
```suggestion
For common queries that you may wish to reuse, you can also create an additional file.
```
I'm not sure that was as helpful as getting to the instructive point sooner.  :smile: It's not as important to explain why, as to model the code that will help the reader be successful.  Can you add an example of a file here, including title, that holds a reusable query? I think that would be a great addition! Then, show a `.astro` file that uses one of these queries.

File: src/content/docs/en/guides/backend/xata.mdx
Comment (sarah11918) [inline src/content/docs/en/guides/backend/xata.mdx:143]:
```suggestion
```
I'm also going to suggest we remove this as not immediately helpful. If you only show approved/recommended usage, then that's much better for the reader. We typically would not include sections that are mostly just a big caution, do not do this.

However, if there is a key difference using Xata with SSR than a pre-rendered page, I think an SSR section makes sense here! If you'd like to show an example of using Xata with SSR, that's perfectly appropriate!

**Review by sarah11918 — COMMENTED** [review]
Thank you so much for your patience, @atilafassina, and I'm glad I could finally give some feedback here!

I'm really looking forward to this guide in the docs to have a little more content on the page, and you can see my comments/suggestions below!

File: src/content/docs/en/guides/backend/xata.mdx
Comment (atilafassina) [reply src/content/docs/en/guides/backend/xata.mdx:21]: Yes. It's not Xata specific. I was trying to follow the lead from the Firebase recipe. I'll add a note to the prerequisites about having the project linked to a DB and pointing to our docs. 


But I'll leave here the `env.d.ts` part because that's Astro specific and I think it could help, wdyt??

File: src/content/docs/en/guides/backend/xata.mdx
Comment (atilafassina) [reply src/content/docs/en/guides/backend/xata.mdx:64]: No no, absolutely right.

I'll just leave the `env.d.ts` part as a reassurance of how to use the env variables better with typescript in an Astro project, if that's alright. Wdyt?

File: src/content/docs/en/guides/backend/xata.mdx
Comment (atilafassina) [reply src/content/docs/en/guides/backend/xata.mdx:129]: My point was just to say "don't make changes to the generated code because once schema updates, your changes will be overwritten".

So how about I say _that_ instead of what I wrote initially?? 😅

**Comment by atilafassina** [comment]
made a few changes @sarah11918 .
I believe this is ready for a second go 😎

**Comment by sarah11918** [comment]
Thanks! Will take another look at this again soon!  :raised_hands:

File: src/content/docs/en/guides/backend/xata.mdx
Comment (ElianCodes) [inline src/content/docs/en/guides/backend/xata.mdx:41]:
```suggestion
```ini title=".xatarc"
{
  "databaseUrl": "https://your-database-url"
}
```
```

**Review by ElianCodes — COMMENTED** [review]
Left a small suggestion, everything else is looking good to me!

File: src/content/docs/en/guides/backend/xata.mdx
Comment (sarah11918) [reply src/content/docs/en/guides/backend/xata.mdx]: Yes, that was my initial thinking! It is Astro specific. Good instinct!

File: src/content/docs/en/guides/backend/xata.mdx
Comment (sarah11918) [reply src/content/docs/en/guides/backend/xata.mdx:97]: Let's give them that reason!

**Review by sarah11918 — COMMENTED** [review]
Looking great! I've made some "editor edits" as a writer, and not as someone who is familiar with Xata. Obviously, my tendency is to CUT in this role, making sentences shorter and more direct, and trying to remove "fluff." But, I don't always know what is fluff and what is essential!

So, I'm going to commit these so we can get the file/page looking fresh as a nice cleanly written document, and then YOU suggest changes where I've gotten things wrong or lost nuance. :sweat_smile:

**Comment by atilafassina** [comment]
> Looking great! I've made some "editor edits" as a writer, and not as someone who is familiar with Xata. Obviously, my tendency is to CUT in this role, making sentences shorter and more direct, and trying to remove "fluff." But, I don't always know what is fluff and what is essential!
> 
> So, I'm going to commit these so we can get the file/page looking fresh as a nice cleanly written document, and then YOU suggest changes where I've gotten things wrong or lost nuance. 😅

Looks perfect to me, the only question is: shouldn't we remove the stub disclaimer at this point?? 🤔

Screenshot 2023-06-09 at 11 58 04

I'm fine either way, just curious.

**Comment by sarah11918** [comment]
Ooh, absolutely! Let's do that, then get this thing meeeeeerrrrrggggeeed!  :smile:

File: src/content/docs/en/guides/backend/xata.mdx
Comment (sarah11918) [inline src/content/docs/en/guides/backend/xata.mdx:6]:
```suggestion
stub: false
```

**Review by sarah11918 — APPROVED** [review]

**Comment by sarah11918** [comment]
Checks have passed! Thanks @atilafassina for your patience and cooperation, and let's gooooo!  :rocket:

**Pull request merged**
